components/post_markdown/post_markdown.jsx

```diff
@@ -63,17 +66,25 @@ export default class PostMarkdown extends React.PureComponent {
         const channelNamesMap = this.props.post && this.props.post.props && this.props.post.props.channel_mentions;
 
         let {message} = this.props;
-        const {post} = this.props;
-        const options = {...this.props.options};
+        const {post, allMentionKeys, mentionKeysWithoutGroups, options, mentionKeysWithoutChannelMentions} = this.props;
 
         this.props.pluginHooks.forEach((o) => {
             if (o && o.hook && post) {
                 message = o.hook(post, message);
             }
         });
 
+        let mentionKeys = [];
         if (post && post.props) {
-            options.mentionHighlight = !post.props.mentionHighlightDisabled;
+            mentionKeys = allMentionKeys;
+
+            if (post.props.disable_group_highlight) {
+                mentionKeys = mentionKeysWithoutGroups;
+            }
+
+            if (post.props.mentionHighlightDisabled) {
+                mentionKeys = mentionKeysWithoutChannelMentions;
```

Comment:
I think this is a slight behaviour change from the previous version since the version with channel-wide mentions disabled may still include group mentions, wouldn't it?